Rebranding (FOG>>PXEL|Host>>Node) & Access Control Feature Enhancement

weitheng10 opened the pull request:
@peirkern-intel review

weitheng10 committed:
Rename fields to support translation, hide fog donation 20220301

weitheng10 committed:
Hide footer bar to simplify the UI
Add "_()" to perfrom localization to translate from:
 FOG to PXEL
 Host to Node
Access Control Plugin Enhancement
 Add new default roles "labUsers"
 Initial data into rules to create default rules for report sub menu, host's submenu and group's submenu
 Initial data into roleRuleAssoc for lab user role to disable module access
 Incease the size of rules.ruleValue from 40 to 100 to cater sub menu module access string
 enhance the logic of deleteSubMenuData
  - able to support submenu removal (can be remove by id dynamically & special characters "&" from report)
  - remove encode/decode of reports module so that we can add a fix the ruleValue (with encode the value will be different for different site)
Added fog_PXEL.css
 - css file use for alter the apperance of web ui

Release Version: 1.5.9.20220302
Baseline Version: 1.5.9

weitheng10 committed:
Release Version: 1.5.9.20220323
Baseline Version: 1.5.9

Enhance Look & Feel Web UI
 - Login Screen
  - Remove Header
  - Change the login screen background
 - Site Header After login
  - change the style to blue icon & grey background
  - Change FOG Project wording to PXEL header logo "pxelhdr.png"
 - change to pxel icon for favicon.ico
 - SUT pxe boot background bg.png to pxel background
 - Fix localization of host to node
  - report module, group membership delete

====================================================================

weitheng10 committed:
Release Version: 1.5.9.20220328
Baseline Version: 1.5.9

Enable access control plugin upon install/reinstall
 add new function "enableAccessControlPlugin" in function.sh call by installfog.sh
 enableAccessControlPlugin using curl command to perform POST request to
 new class activatepluginrun.php allow to call to enable plugin
Add new module About US page
 new page at fogproject\packages\web\lib\pages\aboutuspage.class.php
Enable plugin "PLUGINSYS ENABLED" by default (only for fresh installation will take effect)

====================================================================

peirkern-intel commented on bin/installfog.sh:
```diff
@@ -630,6 +630,7 @@ while [[ -z $blGo ]]; do
                     writeUpdateFile
                     linkOptFogDir
                     updateStorageNodeCredentials
+					enableAccessControlPlugin
```
formatting.

weitheng10 replied:
Fix on next release

peirkern-intel commented on packages/web/lib/fog/page.class.php:
```diff
@@ -457,7 +463,7 @@ public function render()
     {
         global $node;
-        echo '<div class="col-md-3">';
+        echo '<div class="col-md-2">';
```
this is the space after search bar? navbar?

peirkern-intel commented on packages/web/lib/pages/aboutuspage.class.php:
```diff
@@ -0,0 +1,95 @@
+ *
+ * @category AboutUsPage
+ * @package  FOGProject
+ * @author   CES Team <ive.vla.sre.ces@intel.com>
```
author -> WT.

weitheng10 replied:
fixed on next release

peirkern-intel commented on packages/web/lib/pages/aboutuspage.class.php:
```diff
@@ -0,0 +1,95 @@
+ *
+ * @category AboutUsPage
+ * @package  FOGProject
+ * @author   CES Team <ive.vla.sre.ces@intel.com>
```
why we hv two redundant sections?

peirkern-intel commented on packages/web/lib/reports/hosts_and_users.report.php:
```diff
@@ -21,7 +21,7 @@ class Hosts_And_Users extends ReportManagementPage
     public function file()
     {
-        $this->title = _('FOG Host and Users - Search');
+        $this->title = _('FOG Host and Users') .'- ' . _('Search'); 
```
the effect of these changes?

peirkern-intel commented on packages/web/management/activatepluginrun.php:
```diff
@@ -0,0 +1,74 @@
+ * @license  http://opensource.org/licenses/gpl-3.0 GPLv3
+ * @link     https://fogproject.org
+ */
+/**
```
cleanup

weitheng10 replied:
fixed on next release

peirkern-intel commented on packages/web/management/other/index.php:
```diff
@@ -123,7 +135,10 @@
-    echo '<nav class="navbar navbar-inverse navbar-fixed-top">';
+    //CES_CUSTOMIZATION 20220323 START     
+    //echo '<nav class="navbar navbar-inverse navbar-fixed-top">';
+    //echo '<nav class="navbar navbar-fixed-top" style="background-color: #f8f8f8;border-color: #337AB7">';
```
remove

weitheng10 replied:
removed 140 and remain the ori code 139 on next release

weitheng10 committed:
Fix the about us page info & remove printer icon (to avoid double line header)
Fix according to PK code review on code clean up
functions.sh enableAccessControlPlugin need to include noproxy else it cant be execute (tested at KM site)

weitheng10 commented:
Fixed on next release will create another pull request

weitheng10 commented:
@peirkern-intel have complete the fixes kindly review again
